src/middlewared/middlewared/plugins/smart.py

```diff
@@ -296,6 +299,157 @@ class Config:
+        )
+    )
+    async def manual_test(self, disks):
+        verrors = ValidationErrors()
+        if not disks:
+            verrors.add(
+                'manual_test.disks',
+                'Please specify at least one disk.'
+            )
+        else:
+            test_disks_list = []
+            disks_data = await self.middleware.call('disk.query')
+            devices = await camcontrol_list()
+
+            for disk in disks:
+                for d in disks_data:
+                    if disk['identifier'] == d['identifier']:
+                        current_disk = d
+                        test_disks_list.append({
+                            'disk': current_disk['name'],
+                            **disk
+                        })
+                        break
+                else:
+                    verrors.add(
+                        'manual_test.disks.identifier',
+                        f'{disk["identifier"]} is not valid. Please provide a valid disk identifier.'
+                    )
+                    continue
+
+                if disk['alert_service_id'] and not (
+                    await self.middleware.call(
+                        'alertservice.query',
+                        [['id', '=', disk['alert_service_id']]]
+                    )
+                ):
+                    verrors.add(
+                        'manual_test.disks.alert_service_id',
+                        f'Please provide a valid alert_service_id for {disk["identifier"]}'
+                    )
+
+                if current_disk['name'] is None or current_disk['name'].startswith('nvd'):
+                    verrors.add(
+                        'manual_test.disks.identifier',
+                        f'Test cannot be performed for {disk["identifier"]} disk. Failed to retrieve name.'
+                    )
+
+                device = devices.get(current_disk['name'])
+                if not device:
+                    verrors.add(
+                        'manual_test.disks.identifier',
+                        f'Test cannot be performed for {disk["identifier"]}. Unable to retrieve disk details.'
+                    )
+
+        verrors.check()
+
+        smartd_pid = await self.middleware.call('smart.smartd_pid')
+
+        return list(
+            await asyncio_map(functools.partial(self.__manual_test, devices, smartd_pid), test_disks_list, 16)
+        )
+
+    async def __manual_test(self, devices, smartd_pid, disk):
+        device = devices.get(disk['disk'])
+        args = await get_smartctl_args(disk['disk'], device)
+
+        proc = await run(
+            list(
+                filter(bool, ['smartctl', '-t', disk['type'].lower(), '-C' if disk['mode'] == 'FOREGROUND' else None])
+            ) + args,
+            check=False, encoding='utf8'
+        )
+
+        output = {}
+        if proc.returncode:
+            output['error'] = proc.stderr
+            syslog.openlog('smartctl', facility=syslog.LOG_DAEMON)
```

Comment:
Why do we use syslog here?

Reply:
To log `smartctl` command status to `daemon.log` ?
```
root@freenas[~]# grep -rin "smartctl" /var/log/daemon.log 
168:May  9 14:14:27 freenas smartctl: Self test run for /dev/ada0 failed with 4 return code
```

Reply:
Why do we need it there?

Reply:
We don't specifically need it. My thought was that if a user initiates this, he probably wouldn't be checking middlewared logs but daemon log to see what happened with it. Do you think we should perhaps move this to middlewared ? I can have that adjusted

Reply:
User has launched this via UI, why would he want to check system logs for result? Also this brings us to this: I can't find the code that gives us the list of currently running manual tests (and recently finished), is this considered non-necessary?

Reply:
As far as i am aware, the same command `smartctl -l selftest disk` is good for this ? The `smartctl` docs reflect the same. Please let me know if there is something which i am overlooking ?

Reply:
Also, i'll have the logging to `daemon.log` disabled, i think we can just keep that bit to ourselves in middlewared.log then